pkg/cloud/services/autoscaling/autoscalinggroup_test.go

```diff
@@ -220,3 +317,722 @@ func TestService_SDKToAutoScalingGroup(t *testing.T) {
+							sort.Slice(tags, func(i, j int) bool {
+								return *(tags[i].Key) < *(tags[j].Key)
+							})
+						}
+						// sorting tags to avoid failure due to different ordering of tags
+						sortTagsByKey(actual.Tags)
+						sortTagsByKey(expected.Tags)
+						if !reflect.DeepEqual(expected, actual) {
+							t.Fatalf("Actual CreateAutoScalingGroupInput did not match expected, Actual : %v, Expected: %v", actual, expected)
+						}
+						return &autoscaling.CreateAutoScalingGroupOutput{}, nil
+					})
+			},
+			check: func(autoscalinggroup *expinfrav1.AutoScalingGroup, err error) {
+				if err != nil {
+					t.Fatalf("did not expect error: %v", err)
+				}
+				if autoscalinggroup != nil {
+					t.Fatalf("expected asg to be nil but got: %+v", autoscalinggroup)
+				}
+			},
+		},
+		{
+			name:            "should return error if create ASG fails",
+			machinePoolName: "create-asg-fail",
+			expect: func(s *mock_cloudiface.MockClusterScoperMockRecorder, m *mock_autoscalingiface.MockAutoScalingAPIMockRecorder) {
+				s.Name().Times(2).Return("test")
+				s.Info(gomock.Eq("Running instance"))
+				m.CreateAutoScalingGroup(gomock.AssignableToTypeOf(&autoscaling.CreateAutoScalingGroupInput{})).Return(nil, awserrors.NewFailedDependency("dependency failure"))
+				s.Error(gomock.Any(), gomock.Eq("unable to create AutoScalingGroup"))
+			},
+			check: func(autoscalinggroup *expinfrav1.AutoScalingGroup, err error) {
+				if err == nil {
+					t.Fatalf("expected error to occur")
+				}
+				if autoscalinggroup != nil {
+					t.Fatalf("expected asg to be nil but got: %+v", autoscalinggroup)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			asgMock := mock_autoscalingiface.NewMockAutoScalingAPI(mockCtrl)
+			scopeMock := mock_cloudiface.NewMockClusterScoper(mockCtrl)
+			tt.expect(scopeMock.EXPECT(), asgMock.EXPECT())
+			s := &Service{
+				scope:     scopeMock,
+				ASGClient: asgMock,
+			}
+			mps, err := getMachinePoolScope(tt.machinePoolName)
+			if err != nil {
+				t.Fatalf("Failed to create context: %v", err)
+			}
+			asg, err := s.CreateASG(mps)
+			tt.check(asg, err)
+		})
+	}
+}
+
+func TestService_UpdateASG(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tests := []struct {
+		name            string
+		machinePoolName string
+		expect          func(s *mock_cloudiface.MockClusterScoperMockRecorder, m *mock_autoscalingiface.MockAutoScalingAPIMockRecorder)
+		check           func(err error)
+	}{
+		{
+			name:            "should return without error if update ASG is successful",
+			machinePoolName: "update-asg-success",
+			expect: func(s *mock_cloudiface.MockClusterScoperMockRecorder, m *mock_autoscalingiface.MockAutoScalingAPIMockRecorder) {
+				m.UpdateAutoScalingGroup(gomock.AssignableToTypeOf(&autoscaling.UpdateAutoScalingGroupInput{})).Return(&autoscaling.UpdateAutoScalingGroupOutput{}, nil)
+			},
+			check: func(err error) {
```

Comment:
Can we refactor this check func as it is used in multiple places, or directly use it while checking [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/9a86ec7d19650f754cb20882f9b5baa3df802529/pkg/cloud/services/autoscaling/autoscalinggroup_test.go#L552)? Also, its usage is same across UpdateASG, CreateASG tests etc., so you can think of some refactoring